staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go

```diff
@@ -130,6 +137,14 @@ func (c *ConditionController) sync(key string) error {
 		return err
 	}
 
+	// avoid repeated calculation for the same generation
+	c.lastSeenGenerationLock.Lock()
```

Comment:
It seems a read lock here should suffice.

Reply:
true, but this is a slow moving api with an extremely short critical section.  I don't expect significant contention here at all

Reply:
Thanks for the clarification.